pydarn/io/borealis/borealis_formats.py

```diff
@@ -246,27 +353,428 @@ def array_single_element_types(cls):
+                if field in cls.single_element_types():
+                    datatype = cls.single_element_types()[field]
+                    # field is not an array, single element per record.
+                    # unshared_field_dims_site should give empty list.
+                    timestamp_dict[key][field] = datatype(data_dict[field]
+                                                          [record_num])
+                else:  # field in array_dtypes
+                    datatype = cls.array_dtypes()[field]
+                    # need to get the dims correct, not always equal to the max
+                    site_dims = [dimension_function(data_dict, record_num)
+                                 for dimension_function in
+                                 cls.unshared_fields_dims_site()[field]]
+                    index_slice = [slice(0, i) for i in site_dims]
+                    index_slice.insert(0, record_num)
+                    index_slice = tuple(index_slice)
+                    timestamp_dict[key][field] = data_dict[field][index_slice]
+
+        timestamp_dict = cls.flatten_site_arrays(timestamp_dict)
+
+        return timestamp_dict
+
+
+class BorealisRawacfv0_4(BaseFormatClass):
+    """
+    Class containing Borealis Rawacf data fields and their types.
+
+    Rawacf data has been mixed, filtered, and decimated; beamformed and
+    combined into antenna arrays; then autocorrelated and correlated between
+    antenna arrays to produce matrices of num_ranges x num_lags.
+
+    Static Methods
+    --------------
+    is_restructureable: bool
+        Returns True, this format can be restructured to arrays.
+    find_num_ranges(OrderedDict): int
+        Returns num ranges in the data for use in finding dimensions
+    find_num_lags(OrderedDict): int
+        Returns the num lags in the data for use in finding dimensions
+    reshape_site_arrays(OrderedDict): OrderedDict
+        Reshapes the main_acfs, intf_acfs, xcfs fields.
+    """
+
+    @staticmethod
+    def is_restructureable() -> bool:
+        return True
+
+    @staticmethod
+    def find_num_ranges(records: OrderedDict) -> int:
+        """
+        Find the number of ranges given the records dictionary, for
+
+        restructuring to arrays.
+        Num_ranges is unique to a slice so cannot change inside file.
+        """
+        first_key = list(records.keys())[0]
+        num_ranges = records[first_key]['correlation_dimensions'][1]
+        return num_ranges
+
+    @staticmethod
+    def find_num_lags(records: OrderedDict) -> int:
+        """
+        Find the number of lags given the records dictionary, for
+        restructuring to arrays.
+        Num_lags is unique to a slice so cannot change inside file.
+        """
+        first_key = list(records.keys())[0]
+        num_lags = records[first_key]['correlation_dimensions'][2]
+        return num_lags
+
+    @staticmethod
+    def reshape_site_arrays(records: OrderedDict) -> OrderedDict:
```

Comment:
So I feel this abusing inheritance since you are overwriting a lot the methods and they are classmethods or staticmethods. 
Here is a link explaining when to use inheritance: https://www.thoughtworks.com/insights/blog/composition-vs-inheritance-how-choose
Also determining when to use instance, static, and class: https://www.makeuseof.com/tag/python-instance-static-class-methods/